tools/caml_tex.ml

```diff
@@ -19,39 +19,46 @@
 open StdLabels
 open Str
 
-let camlbegin = "\\caml"
-let camlend = "\\endcaml"
-let camlin = {|\\?\1|}
-let camlout = {|\\:\1|}
-let camlbunderline = "\\<"
-let camleunderline = "\\>"
-
-let start newline out s args =
-  Format.fprintf out "%s%s" camlbegin s;
+let camlprefix = "caml"
+let camlin = {|$\\?$\1|}
+let camlout = {|$\\:$\1|}
+let camlbunderline = "<<"
+let camleunderline = ">>"
+
+
+(** Restrict the number of latex environment *)
+type env = Env of string
+let main = Env "example"
+let input_env = Env "input"
+let ok_output = Env "output"
+let error = Env "error"
+let warning = Env "warn"
+let phrase_env = Env ""
+
+
+let start ?(newline=false) out (Env s) args =
```

Comment:
Personally I prefer named arguments over optional arguments or unnamed boolean arguments, so I would use `~newline:bool` in both `start` and `stop` (and probably `code_env` as well).